CONTRIBUTING.md

```diff
@@ -18,18 +21,202 @@ jupyter lab
 
 Step 4) experiment within notebooks with what you wnat the end data scientist experience to be and then merge code in to the codebase as you go.
 
-### Git and Github
+### Learn Git and Github
 
-All our development is done using Git and Github. If you're not too familiar with Git and Github, [start by reviewing this guide](https://guides.github.com/activities/hello-world).
+All our development is done using Git and Github. If you're not too familiar with Git and Github, start by [reviewing this guide](https://guides.github.com/activities/hello-world).
 
 ### Slack
 
 [Join our Slack community](http://slack.openmined.org).
 
+### Issues
+
+On <https://github.com/OpenMined/PySyft/issues> you can find all open Issues. You can find a detailed explanation on how to work with issues below under [Issue Allocation](#issue-allocation).
+
+## Setup
+
+### Forking a Repository
+
+To contribute to PySyft you will need to fork the OpenMind/PySyft repository.
+Then you can work risk-free on your fork.
+
+Fork this Repository. This will create a local copy (Remote Repository) of this repository to your Github profile. Keep a reference to the original project in `upstream` remote.
+```
+$ git clone https://github.com/<your-username>/<repo-name>
+$ cd <repo-name>
+$ git remote add upstream https://github.com/<upstream-owner>/<repo-name>
```

Comment:
Should we use OpenMined for `<upstream-owner>` and PySyft for `<repo-name>` directly? After all, we're writing this in the repo

Reply:
Yes, we can write them down directly, but this will give them the idea of why they are using it.